src/Microsoft.ML.EntryPoints/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,429 @@
+            return new PermutationFeatureImportanceOutput { Metrics = result };
+        }
+    }
+
+    internal sealed class PermutationFeatureImportanceOutput
+    {
+        [TlcModule.Output(Desc = "The PFI metrics")]
+        public IDataView Metrics;
+    }
+
+    internal sealed class PermutationFeatureImportanceArguments : TransformInputBase
+    {
+        [Argument(ArgumentType.Required, HelpText = "The path to the model file", ShortName = "path", Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+        public PredictorModel PredictorModel;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Use feature weights to pre-filter features", ShortName = "usefw", Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+        public bool UseFeatureWeightFilter = false;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Limit the number of examples to evaluate on", ShortName = "numexamples", Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+        public int? NumberOfExamplesToUse = null;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "The number of permutations to perform", ShortName = "permutations", Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+        public int PermutationCount = 1;
+    }
+
+    internal static class PermutationFeatureImportanceUtils
+    {
+        internal static IDataView GetMetrics(
+            IHostEnvironment env,
+            IPredictor predictor,
+            RoleMappedData roleMappedData,
+            PermutationFeatureImportanceArguments input)
+        {
+            IDataView result;
+            if (predictor.PredictionKind == PredictionKind.BinaryClassification)
+                result = GetBinaryMetrics(env, predictor, roleMappedData, input);
+            else if (predictor.PredictionKind == PredictionKind.MulticlassClassification)
+                result = GetMulticlassMetrics(env, predictor, roleMappedData, input);
+            else if (predictor.PredictionKind == PredictionKind.Regression)
+                result = GetRegressionMetrics(env, predictor, roleMappedData, input);
+            else if (predictor.PredictionKind == PredictionKind.Ranking)
+                result = GetRankingMetrics(env, predictor, roleMappedData, input);
+            else
+                throw Contracts.Except(
+                    "Unsupported predictor type. Predictor must be binary classifier, " +
+                    "multiclass classifier, regressor, or ranker.");
+
+            return result;
+        }
+
+        private static IDataView GetBinaryMetrics(
+            IHostEnvironment env,
+            IPredictor predictor,
+            RoleMappedData roleMappedData,
+            PermutationFeatureImportanceArguments input)
+        {
+            var roles = roleMappedData.Schema.GetColumnRoleNames();
+            var featureColumnName = roles.Where(x => x.Key.Value == RoleMappedSchema.ColumnRole.Feature.Value).First().Value;
```

Comment:
>featureColumnName [](start = 16, length = 17)

All these special roles - features, label, group etc. have properties in `RoleMappedSchema`, but you need to check first that they are not null. (here and in the other `Get***Metrics` methods as well).